src/Microsoft.AspNet.Mvc.Core/AntiForgery/AntiForgeryWorker.cs

```diff
@@ -204,6 +189,54 @@ public void Validate([NotNull] HttpContext httpContext, string cookieToken, stri
                 deserializedFormToken);
         }
 
+
+        /// <summary>
+        /// Generates an anti-forgery cookie token and header for this request.
+        /// </summary>
+        /// <param name="context">The HTTP context associated with the current call.</param>
+        public void SetCookieTokenAndHeader([NotNull] HttpContext httpContext)
+        {
+            CheckSSLConfig(httpContext);
+
+            var oldCookieToken = GetCookieTokenNoThrow(httpContext);
+            var newCookieToken = ValidateAndGenerateNewToken(oldCookieToken);
+            
+            SaveCookieTokenAndHeader(httpContext, newCookieToken);
```

Comment:
This method saves cookie and also writes X-frame-options to response header. So I have to call this method even if newcookie is null. I will remove [NotNull] on newCookieToken parameter.